docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1118 @@
+* `ovnkube_controller_admin_network_policies_db_objects`: The total number
+  of OVN NBDB objects (table_name) owned by AdminNetworkPolicy controller in
+  the cluster. `table_name` label here can be either `ACL` or `Address_Set`.
+* `ovnkube_controller_baseline_admin_network_policies_db_objects`: The total
+  number of OVN NBDB objects (table_name) owned by BaselineAdminNetworkPolicy
+  controller in the cluster. `table_name` label here can be either `ACL` or `Address_Set`.
+* `ovnkube_controller_admin_network_policies_rules`: The total number of rules
+  across all admin network policies in the cluster grouped by `direction` and `action`.
+  `direction` can be either `Ingress` or `Egress` and `action` can be either
+  `Pass` or `Allow` or `Deny`.
+* `ovnkube_controller_baseline_admin_network_policies_rules`: The total number of rules
+  across all baseline admin network policies in the cluster grouped by `direction` and `action`.
+  `direction` can be either `Ingress` or `Egress` and `action` can be either
+  `Allow` or `Deny`.
+
+![anp-metrics-1](../../images/anp-metrics-1.png)
+![anp-metrics-2](../../images/anp-metrics-2.png)
+![anp-metrics-3](../../images/anp-metrics-3.png)
+![anp-metrics-4](../../images/anp-metrics-4.png)
+
+## Best Practices and Gotchas!
+
+* Its better to create ANPs at separate priorities so that the precedence
+  is well defined and if you are creating two ANPs at the same priority,
+  then it is the responsibility of the end user to ensure the rules in
+  those ANPs are not overlapping. If it overlaps then its not guaranteed
+  which one will take effect.
+* Unlike ANPs BANP is a singleton object. So there can be only 1 `default`
+  BANP in your cluster. Hence use this policy wisely as a wider default
+  guardrail.
+* API semantics for admin network policies are very different from network
+  policies. There is no implicit deny when you create the policy. You have to
+  explicitly define what you need. Because of this it is not possible to
+  completely isolate the pod from everything using admin network policies for
+  ingress traffic. Only peers you can express for a deny for ingress rule are
+  pods; so atmost only "DENY from all other pods" can be expressed as the best
+  case deny for ingress. In case of egress, one can deny to 0.0.0.0/0 and it will
+  act like a default deny to everything.
+* Unlike network policies, admin network policies are cluster-scoped. So use the
+  empty namespace selector with care because it will select ALL namespaces in your
+  cluster including the infrastructure namespaces like kube-system and kapi-server
+  or openshift-*. Best practice to secure your workloads is to use workload
+  specific labels and then use that as the namespace selector instead.
+* `Pass` is a special case action used to delegate policy decision to the
+  network policy layer. If you don't care about interoperability with existing
+  network policies, then you probably don't need to worry about this feature.
```

Comment:
disagree

Reply:
have other suggestions on rewriting this? delegation is the only use case right?

Reply:
I think we can say that cluster admin **should** care about interoperability with existing network policies, if they have any. And clarify that if some ANP applies to the same connection as Network Policy, ANP's action will be enforced. If that is not desired, a higher-prio ANP with Skip action can be applied.

Reply:
yeah I can rewrite it that way maybe it makes that more clear

Reply:
As above, it's `Allow` that should be the rare special case, not `Pass`:

- "I want to block this traffic" → `Deny`
- "I don't want to block this traffic" → `Pass`
- "I want the pod to be forced to accept this traffic even if it doesn't want it" → `Allow`

Reply:
but there is a section who don't want users to create network policies, they want to control it all with anps.. then they can just use allow or pass and given allow is allow in semantics its easier for them to do that.

Reply:
I have tried to clarify this bit better in docs, let's see if it came out well.

Reply:
> but there is a section who don't want users to create network policies

[citation required]

There are certainly lots of users who would explain what they want as "I want to deny x and y and z, and allow everything else", but I don't think any of them actually mean "`Deny` x and y and z, and `Allow` everything else".

Reply:
anyway, whatever. We can wait and see what sort of questions/confusions we get from customers